main.py

```diff
@@ -1,24 +1,50 @@
-from preprocessing.dataloaders import concat_datasets, CustomDataset, CustomTemporalDataset
+from preprocessing.dataloaders import concat_datasets, SequentialSampler
+import numpy as np
+import torch
 from torch.utils.data import DataLoader
+from utils import set_seed, EarlyStopping, train, test
+import torch.optim as optim
+from models.spacial import BCINet
 import json
 
 
 def main():
     with open("params.json") as fp:
         params = json.load(fp)
 
-    dataset = concat_datasets(params["in_dir"])
-
-    dataLoader = DataLoader(dataset,
-                            batch_size=params['batch_size'],
-                            num_workers=2,
-                            shuffle=True)
+    # Check cuda availability
+    params['device'] = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
Here, I would specify device on params, and in the case needing gpu, check if it is available (else raise an error). I don't think we should modify params throughout the pipeline.

Reply:
It is a common practice to do this in order to have all the parameters in in the same variable. no problem on that

Reply:
I think it's a great idea using a `params.json` and load the parameters into to a dictionary. But this dictionary has to be fixed throughout the pipeline. Modifying it will may only introduce bugs that are harder to track: by adding the `device` field to `params`, we cannot compare the json to the dictionary anymore. I think we can either add a `device` field to the json or create an independent variable device.